gesturecontrol.py

```diff
@@ -7,221 +7,259 @@
+                    vectorZ += 0.0 # Needs work
+                    preX = x + w/2
+                    preY = y + h/2
+
+                    # If unable to acquire the lock, don't update the vectors
+                    if self.vectorLock.acquire(False):
+                        self.updateVectors(vectorX, vectorY, vectorZ)
+                        vectorX = 0.0
+                        vectorY = 0.0
+                        vectorZ = 0.0
+
+                    elapsedTimeSinceFirstDetection = time.time() - startTime
+                    if writecsv:
+                        data = list(self.grabVector())
+                        data.append(elapsedTimeSinceFirstDetection)
+                        vectorDataSet.append(data)
+
+                # update the FPS counter
+                fps.update()
+                fps.stop()
+
+                # initialize the set of information we'll be displaying on
+                # the frame
+                info = [
+                    ("Tracker", self.trackerType),
+                    ("Success", "Yes" if success else "No"),
+                    ("FPS", "{:.2f}".format(fps.fps())),
+                    ("Centre of Box", "{}: {}".format(x + w/2, y + h/2)),
+                    ("Vector of Box", "{:.2f}, {:.2f}, {:.2f}".format(self.vectorX, self.vectorY, self.vectorZ)),
+                ]
+
+                # loop over the info tuples and draw them on our frame
+                for (i, (k, v)) in enumerate(info):
+                    text = "{}: {}".format(k, v)
+                    cv2.putText(frame, text, (10, H - ((i * 20) + 20)),
+                        cv2.FONT_HERSHEY_SIMPLEX, 0.6, (0, 0, 255), 2)
+
+            # show the output frame
+            cv2.imshow("Frame", frame)
+            key = cv2.waitKey(1) & 0xFF
+
+            # if the 's' key is selected, we are going to "select" a bounding
+            # box to track
+            if key == ord("s"):
+                # select the bounding box of the object we want to track (make
+                # sure you press ENTER or SPACE after selecting the ROI)
+                initBB = cv2.selectROI("Frame", frame, fromCenter=False,
+                    showCrosshair=True)
+                startTime = time.time()
+
+                # start OpenCV object tracker using the supplied bounding box
+                # coordinates, then start the FPS throughput estimator as well
+                self.tracker.init(frame, initBB)
+                fps = FPS().start()
+
+            # if the `q` key was pressed, break from the loop
+            elif key == ord("q"):
+                break
+
+        # if we are using a webcam, release the pointer
+        vs.stop()
+        self.running = False
+
+        # close all windows
+        cv2.destroyAllWindows()
+
+        if writecsv:
+            self.createCSV(vectorDataSet)
+
+    def playback(self, fileName):
+        dataList = self.getCSVData(fileName)
+        if dataList == []:
```

Comment:
Initially I decided not to put an error check here because I would have had an empty list if the file was empty (and the for loop would just be ignored).